sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1418,6 +1449,51 @@ def process_bundle(self, inputs, expected_outputs, parallel_uid_counter=None):
 
     return result, split_results
 
+class ParallelBundleManager(BundleManager):
+  _uid_counter = 0
+  def process_bundle(self, inputs, expected_outputs):
+    input_value = list(inputs.values())[0]
```

Comment:
```
data_input[transform.unique_name] = pcoll_buffers[pcoll_id]
pcoll_buffers[buffer_id] = _GroupingBuffer(
              pre_gbk_coder, post_gbk_coder, windowing_strategy, self._num_workers)
```
We put only one buffer to each input, so taking the first one only is same as taking all.

Reply:
This may not be the case in the future. If we want to make this assumption, we should at least assert it. But preferable IMHO to be general as that's not too hard.

Reply:
Today I found a case where we shouldn't split inputs, which is a transform with timer and when use grpc handler. I would like to get your advice how to know when we should split inputs and when we shouldn't. I added a comment with examples at the new PR.

Reply:
I found a way to work around by moving type check before entering to for loop. Fixed at the new PR.